### PR TITLE
Refactor/pcc common code

### PR DESCRIPTION
**Pull request by ganeshkumar1989**
| Question         | Answer
| ---------------- | ---
| Branch?          | `develop`
| Bug fix?         | yes
| New feature?     | no
| Breaking change? | yes
| Tickets          | Fix #MANAGER-5672
| License          | BSD 3-Clause

## Description

Refactor and move pcc as a set of components so that the code can be reused for Essentials (MANAGER-5456)

### CODE REVIEW

**Comment by ovh-cds** [comment]
CDS Report test-manager_2#15410.0 ✘  
* Stage 1
  * lint ✔ 
  * SemVer ✘

**Comment by marie-j** [comment]
@ganeshkumar1989 looks like some commits doesn't belong to this PR, a rebase might be needed
Thanks

**Comment by ganeshkumar1989** [comment]
> @ganeshkumar1989 looks like some commits don't belong to this PR, a rebase might be needed
> Thanks

@marie-j , We had the HPC changes on 2 other PRs, which had to be picked up before the refactoring started:
https://github.com/ovh/manager/pull/3464
https://github.com/ovh/manager/pull/3473
These changes are on PCC and would cause conflicts if not picked up

**Comment by ovh-cds** [comment]
CDS Report installation#15417.0 ✘  
* Stage 1
  * Install ✘

**Comment by ovh-cds** [comment]
CDS Report installation#15418.0 ✘  
* Stage 1
  * Install ✘

**Comment by ovh-cds** [comment]
CDS Report installation#15418.1 ✘  
* Stage 1
  * Install ✘

**Comment by ovh-cds** [comment]
CDS Report installation#15418.2 ✘  
* Stage 1
  * Install ✘

**Comment by ovh-cds** [comment]
CDS Report installation#15422.0 ✘  
* Stage 1
  * Install ✘

**Comment by ovh-cds** [comment]
CDS Report installation#15422.1 ✘  
* Stage 1
  * Install ✘

**Comment by ovh-cds** [comment]
CDS Report installation#15436.0 ✘  
* Stage 1
  * Install ✘

**Comment by ovh-cds** [comment]
CDS Report installation#15463.0 ✘  
* Stage 1
  * Install ✘

**Comment by ovh-cds** [comment]
CDS Report installation#15474.0 ✘  
* Stage 1
  * Install ✘

**Review by Jisay — DISMISSED** [review]

File: packages/manager/apps/dedicated/client/app/app.js [hunk 1]
```diff
@@ -258,6 +247,7 @@ angular
       });
 
       $state.defaultErrorHandler((error) => {
+        console.log('error', error);
```
Comment (marie-j) [inline packages/manager/apps/dedicated/client/app/app.js:250]:
```suggestion
```

File: packages/manager/apps/dedicated/client/app/components/dedicated-cloud/dashboard/tiles/options/index.js [hunk 2]
```diff
@@ -0,0 +1,33 @@
+import activationStatus from './components/activation-status';
+import descriptionsBasicOption from './descriptions/basic-option';
+import descriptionsCertification from './descriptions/certification';
+import order from './models/order';
+import servicePack from '../../../service-pack';
+// import upgrade from '../../../../../dedicatedCloud/service-pack/upgrade';
```
Comment (marie-j) [inline packages/manager/apps/dedicated/client/app/components/dedicated-cloud/dashboard/tiles/options/index.js:6]:
```suggestion

```

**Comment by ovh-cds** [comment]
CDS Report installation#15514.0 ✘  
* Stage 1
  * Install ✘

File: packages/manager/apps/dedicated/client/app/app.js (hunk 1 above)
Comment (ganeshkumar1989) [reply packages/manager/apps/dedicated/client/app/app.js]: This is removed :)

File: packages/manager/apps/dedicated/client/app/components/dedicated-cloud/dashboard/tiles/options/index.js (hunk 2 above)
Comment (ganeshkumar1989) [reply packages/manager/apps/dedicated/client/app/components/dedicated-cloud/dashboard/tiles/options/index.js]: This is removed. Thanks :)

**Review by radireddy — APPROVED** [review]

**Comment by ovh-cds** [comment]
CDS Report installation#15539.0 ✘  
* Stage 1
  * Install ✘

File: packages/manager/apps/dedicated/client/app/app.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (ganeshkumar1989) [reply packages/manager/apps/dedicated/client/app/app.js]: done :)

**Comment by ovh-cds** [comment]
CDS Report installation#15588.0 ✘  
* Stage 1
  * Install ✘

**Comment by ovh-cds** [comment]
CDS Report installation#15614.0 ✘  
* Stage 1
  * Install ✘

**Comment by ovh-cds** [comment]
CDS Report installation#15623.0 ✘  
* Stage 1
  * Install ✘

**Comment by ovh-cds** [comment]
CDS Report installation#15636.0 ✘  
* Stage 1
  * Install ✘

**Comment by ovh-cds** [comment]
CDS Report installation#15638.0 ✘  
* Stage 1
  * Install ✘

**Comment by ovh-cds** [comment]
CDS Report installation#15687.0 ✘  
* Stage 1
  * Install ✘

**Comment by ovh-cds** [comment]
CDS Report installation#16110.0 ✘  
* Stage 1
  * Install ✘

**Comment by ovh-cds** [comment]
CDS Report installation#16115.0 ✘  
* Stage 1
  * Install ✘

**Comment by ovh-cds** [comment]
CDS Report installation#16149.0 ✘  
* Stage 1
  * Install ✘

**Comment by ovh-cds** [comment]
CDS Report test#16157.0 ✘  
* Stage 1
  * lint ✘  
  * SemVer ✔

**Comment by ovh-cds** [comment]
CDS Report test#16174.0 ✘  
* Stage 1
  * lint ✘  
  * SemVer ✔

**Comment by ovh-cds** [comment]
CDS Report test#16506.0 ✘  
* Stage 1
  * lint ✘  
  * SemVer ✔

**Comment by ovh-cds** [comment]
CDS Report test#16516.0 ✘  
* Stage 1
  * lint ✘  
  * SemVer ✔

**Review by antleblanc — APPROVED** [review]

**Pull request merged**